agent/config/config.go

```diff
@@ -742,11 +742,16 @@ type ACL struct {
 }
 
 type Tokens struct {
-	Master      *string `mapstructure:"master"`
-	Replication *string `mapstructure:"replication"`
+	InitialManagement *string `mapstructure:"initial_management"`
+	Replication       *string `mapstructure:"replication"`
+	AgentRecovery     *string `mapstructure:"agent_recovery"`
+	Default           *string `mapstructure:"default"`
+	Agent             *string `mapstructure:"agent"`
+
+	// DEPRECATED (ACL) - renamed to "initial_management"
+	Master *string `mapstructure:"master"`
+	// DEPRECATED (ACL) - renamed to "agent_recovery"
```

Comment:
It doesn't work to put these into the `DeprecatedConfig` because they are nested structs, is that right?

We could do something like embed a `deprecated ACLTokens` type and move the fields to that type. Or maybe a bit easier, maybe we could rename these fields to `DeprecatedMaster` and `DeprecatedAgentMaster`? Since we have mapstructure names for them I think the field names don't matter.

Reply:
Yep, I tried adding a nested `ACL` struct to `DeprecatedConfig` but it didn't look like mapstructure could correctly split the assignments between fields on this and the proper `ACL` struct on `Config`:

```Go
type DeprecatedConfig struct {
	ACL struct {
		Tokens struct {
			Master *string `mapstructure:"master"`
		} `mapstructure:"tokens"`
	} `mapstructure:"acl"`
}
```

```
* invalid config key acl.default_policy
* invalid config key acl.down_policy
* invalid config key acl.token_ttl
* invalid config key acl.policy_ttl
```

I think splitting the deprecated fields out into an embedded struct would make sense 👍🏻